apps/vmq_diversity/src/vmq_diversity_cache.erl

```diff
@@ -346,12 +364,22 @@ key(MP, ClientId) when is_binary(MP)
 
 match(Key, Input) ->
     Type = type(Input),
-    case ets:lookup(table(cache), Key) of
+    case ets:select(table(cache), ?ms(Key)) of
         [] -> no_cache;
         [{_, PubAclHashes, _}] when Type == publish ->
             match_(table(Type), Input, PubAclHashes);
         [{_, _, SubAclHashes}] when Type == subscribe ->
-            match_(table(Type), Input, SubAclHashes)
+            match_(table(Type), Input, SubAclHashes);
+        Elems ->
```

Comment:
as this is the generalization of the two case clauses above, I wonder if we should replace them by the generalization. I guess lists:reverse on a one-element list is free.

Reply:
Alternatively, one could use `ets:select_reverse`  instead of the extra `lists:reverse`

Reply:
Right, I wasn't thinking it through when trying to make this efficient. Since `ets:select_reverse` exists (didn't know that, thanks), I simplified this a bit further, as I could simply ask for a single element.